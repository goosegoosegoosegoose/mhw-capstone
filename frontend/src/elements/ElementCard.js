import React from "react";
import { Card } from "react-bootstrap";
import { useNavigate } from "react-router-dom";

const ElementCard = ({ele, img}) => {
  const nav = useNavigate();

  const handleClick = () => {
    nav(`/elements/${ele}`)
  }

  return (
    <Card className="my-3 mx-3 col-lg-2" style={{cursor: "pointer"}} onClick={handleClick} bg="dark">
      <div className="row justify-content-center">
        <div className="col-sm-8 pt-2 mt-2">
          {img ? <Card.Img variant="top" src={img} alt="element" /> : <></>}
        </div>
      </div>
      <Card.Body>
        <Card.Title>{(ele ? ele.toUpperCase() : null)}</Card.Title>
      </Card.Body>
    </Card>
  )
}

export default ElementCard;