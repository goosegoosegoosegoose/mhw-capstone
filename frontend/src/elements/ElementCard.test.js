import { render } from '@testing-library/react';
import ElementCard from "./ElementCard";
import { MemoryRouter } from 'react-router-dom';

test('smoke test', () => {
  render(
    <MemoryRouter>
      <ElementCard />
    </MemoryRouter>
  );
});

test('snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <ElementCard />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});