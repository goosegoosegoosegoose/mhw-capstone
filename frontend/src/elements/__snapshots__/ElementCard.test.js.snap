// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-3 mx-3 col-lg-2 card bg-dark"
    style="cursor: pointer;"
  >
    <div
      class="row justify-content-center"
    >
      <div
        class="col-sm-8 pt-2 mt-2"
      />
    </div>
    <div
      class="card-body"
    >
      <div
        class="card-title h5"
      />
    </div>
  </div>
</DocumentFragment>
`;
